dango/lending/src/query.rs

```diff
@@ -64,3 +77,62 @@ pub fn query_debts(
         .take(limit as usize)
         .collect()
 }
+
+pub fn query_collateral_powers(
+    storage: &dyn Storage,
+) -> StdResult<BTreeMap<Denom, CollateralPower>> {
+    COLLATERAL_POWERS.load(storage)
+}
+
+/// Calculates the health of a margin account.
+pub fn calculate_account_health(
+    querier: &QuerierWrapper,
+    margin_account: Addr,
+    debts: Coins,
+    collateral_powers: BTreeMap<Denom, CollateralPower>,
+) -> anyhow::Result<HealthResponse> {
+    // Calculate the total value of the debts.
+    let mut total_debt_value = Udec128::ZERO;
+    for coin in debts {
+        let price = dango_oracle::raw_query_price(querier, &coin.denom)?;
+        total_debt_value += price.value_of_unit_amount(coin.amount);
+    }
+
+    // Calculate the total value of the account's collateral adjusted for the collateral power.
+    let mut total_adjusted_collateral_value = Udec128::ZERO;
+    for (denom, power) in collateral_powers {
+        let collateral_balance = querier.query_balance(margin_account, denom.clone())?;
+
+        // As an optimization, don't query the price if the collateral balance is zero.
+        if collateral_balance.is_zero() {
+            continue;
+        }
+
+        let price = dango_oracle::raw_query_price(querier, &denom)?;
+        let collateral_value = price.value_of_unit_amount(collateral_balance);
+        total_adjusted_collateral_value += collateral_value * power.into_inner();
+    }
+
+    if total_adjusted_collateral_value.is_zero() {
+        bail!("The account has no collateral");
+    }
+
+    // Calculate the utilization rate.
+    let utilization_rate = total_debt_value / total_adjusted_collateral_value;
```

Comment:
for this you can just do `.checked_div`

Reply:
Do we prefer to use checked math ops always? Any pros and cons to consider here? I'm actually not sure why checked ops are recommended in CosmWasm. Cause actually non-checked versions give a much better backtrace with line number.